sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -203,7 +204,9 @@ public static void main(
       JvmInitializers.runBeforeProcessing(options);
 
       LOG.info("Entering instruction processing loop");
-      control.processInstructionRequests(options.as(GcsOptions.class).getExecutorService());
+      ExecutorService executorService = options.as(GcsOptions.class).getExecutorService();
```

Comment:
This does not come from this PR, but shouldn't this be in a more general options type than `GcsOptions` ?

Reply:
Also shouldn't a change in the harness trigger other tests to validate that we don't break portability (Flink PVR for example).

Reply:
agreed on both points. will have to run spark and flink pvr tests manually.